addons/mrp_account/models/account_move.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+from collections import defaultdict
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _get_invoiced_qty_per_product(self):
+        # Replace the kit-type products with their components
+        qties = defaultdict(float)
+        res = super()._get_invoiced_qty_per_product()
+        for product, qty in res.items():
+            bom_kit = self.env['mrp.bom']._bom_find(product=product, company_id=self.company_id[:1].id, bom_type='phantom')
+            if bom_kit:
+                invoiced_qty = product.uom_id._compute_quantity(qty, bom_kit.product_uom_id, round=False)
+                factor = invoiced_qty / bom_kit.product_qty
+                dummy, bom_sub_lines = bom_kit.explode(product, factor)
+                for bom_line, bom_line_data in bom_sub_lines:
+                    qties[bom_line.product_id] += bom_line.product_uom_id._compute_quantity(bom_line_data['qty'], bom_line.product_id.uom_id)
+            else:
+                qties[product] += qty
```

Comment:
in case you install mrp, deliver and invoice a non-kit product. `_get_invoiced_qty_per_product` will get the quantity in `super` then again here with the `+=` ?

Reply:
`qties` is a new dict so it will just copy the quantity of the non-kit product
I did this to keep the non-kit products and replace the kit-product by its components
I could rather use `res` and something like `del res[product]` directly in the if-block

Reply:
I think it's better like this
What I said in the previous message is wrong, we can't delete the product from `res` while iterating on it
And setting `res[product] = 0` may be confusing IMHO (it could be interpreted as "not invoiced yet" for instance)